skore/src/skore/_sklearn/_estimator/feature_importance_accessor.py

```diff
@@ -532,9 +581,54 @@ def _feature_permutation(
         if cache_key in self._parent._cache:
             score = self._parent._cache[cache_key]
         else:
+            if not isinstance(self._parent.estimator_, Pipeline) or at_step == 0:
+                estimator = self._parent.estimator_
+                X_transformed = X_
+                if hasattr(estimator, "feature_names_in_"):
+                    feature_names = estimator.feature_names_in_
+                elif hasattr(estimator, "n_features_in_"):
+                    feature_names = [
+                        f"Feature #{i}" for i in range(estimator.n_features_in_)
+                    ]
+                else:
+                    feature_names = [
+                        f"Feature #{i}" for i in range(X_transformed.shape[1])
+                    ]
+
+            else:
+                pipeline = self._parent.estimator_
+                if not isinstance(at_step, str | int):
+                    raise ValueError(
+                        f"at_step must be an integer or a string; got {at_step!r}"
+                    )
+
+                if isinstance(at_step, str):
+                    # Make at_step an int and process it as usual
+                    at_step = list(pipeline.named_steps.keys()).index(at_step)
+
+                if isinstance(at_step, int):
+                    if abs(at_step) >= len(pipeline.steps):
+                        raise ValueError(
+                            "at_step must be strictly smaller in magnitude than the "
+                            "number of steps in the Pipeline, which is "
+                            f"{len(pipeline.steps)}; got {at_step}"
+                        )
+                    feature_eng, estimator = pipeline[:at_step], pipeline[at_step:]
+                    X_transformed = feature_eng.transform(X_)
+
+                if hasattr(estimator, "feature_names_in_"):
+                    feature_names = estimator.feature_names_in_
```

Comment:
From the coverage report, it seems that we don't cover the following two lines: 620, 624. Could you amend the test to check those as well.